### ISSUE TITLE
Query analyze/visualize gives no feedback, allows multiple parallel executions.

### ISSUE THREAD

**Issue by fake-name** [ankane/pghero#101]
I have a very expensive query I was experimenting with (general execution time is ~1000 seconds).

If you use the Explain/Analyze or Explain/Visualize interface, clicking the "Analyze" or "Visualize" buttons executes an `EXPLAIN ANALYZE` in the background, silently, with no feedback *for every click of the button*.

I wound up with about 10 instances of my hugely expensive query running concurrently because I assumed I missed the button, and clicked it a number of times.

It seems like the analyze button should probably block/be-disabled until the query has either finished, or been cancelled. 

------

In other news, I typoed `pg_terminate_backend()` as `pg_germinate_backend()`  while cancelling the queries, and my server now has a lawn.

**Comment by ankane** [comment]
Glad to see there was a bit of humor with it. I added a note about potential danger as a first step. I like the idea of tracking the query, as right now, if the query goes beyond your web server timeout, you have to find it in long running queries and kill it. Another idea would be to use `SET LOCAL` to add a statement timeout to prevent this, but then you couldn't explain long queries.

**Comment by fake-name** [comment]
Yeah, I initially forgot I could kill queries through the web interface (and I had a `psql` console open anyways, so I had killed the queries manually before I navigated back to the web-interface root (whoops)).

This is definitely more of a wishlist thing then a actual issue, at the moment.

------

How does the callback that returns the results of the button-presses work? While it wouldn't handle a page-refresh or close+reopen, it'd be a nice 90% solution to disable the button until the EXPLAIN/ANALYZE/etc... request returns.
